assets/src/hooks/useNotificationsReducer.ts

```diff
@@ -107,41 +107,47 @@ export type Dispatch = ReactDispatch<Action>
 type Reducer = (state: State, action: Action) => State
 
 export const notificationsReducer = (
-  notifications: Notification[],
+  notifications: Notification[] | null,
   action: Action
-): Notification[] => {
+): Notification[] | null => {
   switch (action.type) {
     case "ADD_NOTIFICATION": {
       const newNotification = (action as AddNotificationAction).payload
         .notification
-      return [newNotification, ...notifications]
+      return [newNotification, ...(notifications || [])]
     }
     case "EXPIRE_NOTIFICATIONS":
-      return notifications.filter((notification) => {
-        const maxAgeInMs = 8 * 60 * 60 * 1000
-        const now = (action as ExpireNotificationsAction).payload.now
-        const ageInMs = now.valueOf() - notification.createdAt.valueOf()
-        return ageInMs < maxAgeInMs
-      })
+      return notifications
+        ? notifications.filter((notification) => {
+            const maxAgeInMs = 8 * 60 * 60 * 1000
+            const now = (action as ExpireNotificationsAction).payload.now
+            const ageInMs = now.valueOf() - notification.createdAt.valueOf()
+            return ageInMs < maxAgeInMs
+          })
+        : null
     case "MARK_ALL_AS_READ":
-      return notifications.map((notification) => ({
-        ...notification,
-        state: "read",
-      }))
+      return notifications
+        ? notifications.map((notification) => ({
+            ...notification,
+            state: "read",
+          }))
+        : null
     case "SET_NOTIFICATIONS": {
       return (action as SetNotificationsAction).payload.notifications
     }
     case "TOGGLE_READ_STATE": {
       const notificationToToggle = (action as ToggleReadStateAction).payload
         .notification
-      return notifications.map((notification) =>
-        notification.id === notificationToToggle.id
-          ? {
-              ...notification,
-              state: otherNotificationReadState(notification.state),
-            }
-          : notification
-      )
+      return notifications
+        ? notifications.map((notification) =>
+            notification.id === notificationToToggle.id
+              ? {
+                  ...notification,
+                  state: otherNotificationReadState(notification.state),
+                }
+              : notification
+          )
+        : null
```

Comment:
suggestion(non-blocking): because `notifications` is possibly `null`, if you use the `&&` operator instead of the ternary operator, I think it should return `null` if `notifications` is `null`?

But if you prefer to be explicit with the ternary operator I'm fine with that!

Reply:
Thanks for the suggestion, updated in [4e15e57](https://github.com/mbta/skate/pull/2021/commits/4e15e5750a0dd621a7cbd89439398529983a88bb)